mono/mini/mini-exceptions.c

```diff
@@ -3369,7 +3369,7 @@ mono_handle_native_crash (const char *signal, MonoContext *mctx, MONO_SIG_HANDLE
 void
 mono_handle_native_crash (const char *signal, MonoContext *mctx, MONO_SIG_HANDLER_INFO_TYPE *info)
 {
-	g_assert_not_reached ();
```

Comment:
I think `MONO_CROSS_COMPILE` doesn't mean that the build and host platforms are different.  It means the host and target platforms are different: e.g we're building an OSX amd64 AOT compiler that can target iOS arm64.  So in that case we don't expect to run any managed code on the host, so we don't want to use our own crash handler.

Something like that.

So I think we still want `g_assert_not_reached` here for `MONO_CROSS_COMPILE`.  And for `DISABLE_CRASH_REPORTING` we want a `return` (or possibly `mono_post_native_crash_handler (signal, mctx, info, mono_do_crash_chaining)` - I'm not sure what that does exactly)

Reply:
That was my understanding of `MONO_CROSS_COMPILE` also, but cannot see how that categorically excludes the AOT compiler itself from crashing.

Reply:
Like @vargaz mentioned (if I understand "old crash handler" correctly), perhaps the #ifndef there was wrong in the first place.